helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -0,0 +1,253 @@
+    _record.setSimpleField(CloudConfigProperty.CLOUD_INFO_PARSE_CLASS_NAME.name(),
+        cloudInfoParserClassName);
+  }
+
+  /**
+   * Get the CLOUD_INFO_PARSE_CLASS_NAME field.
+   * @return CLOUD_INFO_PARSE_CLASS_NAME field.
+   */
+  public String getCloudInfoParserClassName() {
+    return _record.getSimpleField(CloudConfigProperty.CLOUD_INFO_PARSE_CLASS_NAME.name());
+  }
+
+  public static class Builder {
+    private String _clusterName = null;
+    private boolean _cloudEnabled = DEFAULT_CLOUD_ENABLED;
+    private String _cloudID;
+    private List<String> _cloudInfoURLList;
+    private String _cloudInfoParserClassName;
+
+    public CloudConfig build() {
+      validate();
+      return new CloudConfig(_clusterName, _cloudEnabled, _cloudID, _cloudInfoURLList, _cloudInfoParserClassName);
+    }
+
+    public Builder() {
+    }
+
+    public Builder(String clusterName) {
+      _clusterName = clusterName;
+    }
+
+    public Builder(CloudConfig cloudConfig) {
+      _clusterName = cloudConfig.getClusterName();
+      _cloudEnabled = cloudConfig.getCloudEnabled();
+      _cloudID = cloudConfig.getCloudID();
+      _cloudInfoURLList = cloudConfig.getCloudInfoURL();
+      _cloudInfoParserClassName = cloudConfig.getCloudInfoParserClassName();
+    }
+
+    public Builder setClusterName (String v) {
+      _clusterName = v;
+      return this;
+    }
+
+    public Builder setCloudEnabled (boolean isEnabled) {
+      _cloudEnabled = isEnabled;
+      return this;
+    }
+
+    public Builder setCloudID (String v) {
+      _cloudID = v;
+      return this;
+    }
+
+    public Builder setCloudInfoURL (List<String> v) {
+      _cloudInfoURLList = v;
+      return this;
+    }
+
+    public Builder setCloudInfoURLField (String v) {
+      if (_cloudInfoURLList == null) {
+        _cloudInfoURLList = new ArrayList<String>();
+      }
+      _cloudInfoURLList.add(v);
+      return this;
+    }
+
+    public Builder setCloudInfoParserClassName (String v) {
+      _cloudInfoParserClassName = v;
+      return this;
+    }
+
+
+    public String getClusterName () {
+      return _clusterName;
+    }
+
+    public boolean getCloudEnabled () {
+      return _cloudEnabled;
+    }
+
+    public String getCloudID () {
+      return _cloudID;
+    }
+
+    public List<String> getCloudInfoURL () {
+      return _cloudInfoURLList;
+    }
+
+    public String getCloudInfoParserClassName () {
+      return _cloudInfoParserClassName;
+    }
+
+
+    private void validate() {
+      if (_clusterName == null) {
+        throw new HelixException("The cluster name/ID cannot be null.");
+      }
+
+      if (_cloudEnabled) {
+        if (_cloudID == null || _cloudInfoURLList == null || _cloudInfoParserClassName == null || _cloudInfoURLList.isEmpty()) {
+          throw new HelixException(
+              "This Cloud Configuration is Invalid. The configuration must have all of the fields if enabled.");
```

Comment:
This could be confusing here. How the user know the all fields?

Reply:
Addressed. Thanks.